syntax = "proto3";

// Get protoc here https://github.com/protocolbuffers/protobuf/releases
// .\protoc --python_out=. --pyi_out=. CAP.proto

message ActorInfo {
    string name = 1;
    optional string namespace = 2;
    optional string description = 3;
}

message ActorRegistration {
    ActorInfo actor_info = 1;
}

message ActorLookup {
    optional ActorInfo actor_info = 1;
    // TODO:  May need more structure here for semantic service discovery
    // optional string service_descriptor = 2;
}

message ActorInfoCollection {
    repeated ActorInfo info_coll = 1;
}

message ActorLookupResponse {
    bool found = 1;
    optional ActorInfoCollection actor = 2;
}

message Ping {
}

message Pong {
}
